app/src/main/java/io/appium/uiautomator2/handler/GetPackages.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.handler;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static io.appium.uiautomator2.model.Session.NO_ID;
+
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.appium.uiautomator2.handler.request.NoSessionCommandHandler;
+import io.appium.uiautomator2.handler.request.SafeRequestHandler;
+import io.appium.uiautomator2.http.AppiumResponse;
+import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.model.api.touch.appium.PackageModel;
+
+public class GetPackages extends SafeRequestHandler implements NoSessionCommandHandler {
+    public GetPackages(String mappedUri) {
+        super(mappedUri);
+    }
+
+    @Override
+    protected AppiumResponse safeHandle(IHttpRequest request) {
+        List<PackageModel> appDetails = new ArrayList<PackageModel>();
+        PackageManager manager = getApplicationContext().getPackageManager();
+        List<ApplicationInfo> apps = manager.getInstalledApplications(manager.GET_META_DATA);
+        for (ApplicationInfo appInfo : apps) {
+            // Filtering out unnecessary sub packages without Intent
+            if (manager.getLaunchIntentForPackage(appInfo.packageName) != null) {
+                appDetails.add(new PackageModel(appInfo.packageName,
+                        manager.getLaunchIntentForPackage(appInfo.packageName).getComponent().getClassName(),
```

Comment:
What happen when `getLaunchIntentForPackage(appInfo.packageName)` is null, or `getComponent` is null. They are nullable.

Reply:
They both are Nullable, if an application has an intent then for sure it will have a component, so application with no component will automatically get filter out from the 46 condition and also getComponent is also Nullable

Reply:
ah, missed here called the method twice.
This endpoint tries to show installed packages and each information, then maybe if would be nice to keep the package name (since it exists) but no activities rather than filtering the package out entirely. What do you think?
So, the user can understand some packages exist in the device but they has no launchable activities by this endpoint.

Reply:
The intend was to show all the main applications since the package without Intent is just a subpackage of it's parent package which can just create confusion to the user and actually it's pretty much useless, if later on anyone needs that then surely we can modify accordingly but i feel it's unnecessary right now to show?

Reply:
Then, could you leave the intention as docstring in this class for future maintenance?
`GetPackages` made me think the app package is primary, then app activities etc as additional information. The current intention is to get app activities information than its package (since filter by package).

Reply:
Yea, could you leave what kind of activities will be available, and no activities packages won't be available?
I mean not a comment for the implementation, just this method's entire behavior as the module's description. Also, it wole be appreciate to add this method uses `android.permission.QUERY_ALL_PACKAGES` here, or a comment in the manifest to link each

Reply:
done

Reply:
Sorry, missing commit?

Reply:
Done now